pkg/lightning/restore/restore.go

```diff
@@ -923,7 +934,20 @@ func (rc *Controller) saveStatusCheckpoint(tableName string, engineID int32, err
 		metric.RecordEngineCount(statusIfSucceed.MetricName(), err)
 	}
 
-	rc.saveCpCh <- saveCp{tableName: tableName, merger: merger}
+	waitCh := make(chan error, 1)
+	rc.saveCpCh <- saveCp{tableName: tableName, merger: merger, waitCh: waitCh}
+
+	var saveCpErr error
+	select {
+	case saveCpErr = <-waitCh:
+	case <-ctx.Done():
+		saveCpErr = ctx.Err()
```

Comment:
I think we can just return `ctx.Err()` here to avoid output an error log with context canceled error since it is useless and annoying